ddp-workspace/projects/ddp-pancan/src/app/components/activity/activity.component.ts

```diff
@@ -16,7 +23,37 @@ type Block = ActivitySection['blocks'][number]
     templateUrl: './activity.component.html',
     providers: [SubmitAnnouncementService, SubmissionManager],
 })
-export class ActivityComponent extends ActivityRedesignedComponent {
+export class ActivityComponent extends ActivityRedesignedComponent implements OnInit, OnDestroy {
+    dialogsClosedSubscription: Subscription;
+
+    constructor(
+        public editDialog: MatDialog,
+        logger: LoggingService,
+        windowRef: WindowRef,
+        changeRef: ChangeDetectorRef,
+        renderer: Renderer2,
+        submitService: SubmitAnnouncementService,
+        analytics: AnalyticsEventsService,
+        participantsSearch: ParticipantsSearchServiceAgent,
+        @Inject(DOCUMENT) document: any,
+        injector: Injector
+    ) {
+        super(logger, windowRef, changeRef, renderer, submitService, analytics, participantsSearch, document, injector);
+    }
+
+    public ngOnInit(): void {
+        super.ngOnInit();
+
+        this.dialogsClosedSubscription = this.editDialog.afterAllClosed.subscribe(() => {
+            this.showEmptyNestedActivityError = !this.hasNoEmptyNestedActivity;
+            setTimeout(() => this.showEmptyNestedActivityError = !this.hasNoEmptyNestedActivity, 200); // quick and dirty solution to re-check, in case activity was deleted a bit later
+        });
+    }
+
+    public ngOnDestroy(): void {
+        this.dialogsClosedSubscription.unsubscribe();
```

Comment:
No need to unsubscribe from afterClosed() as it auto completes itself. 
See [Angular material dialog](https://github.com/angular/components/blob/ae41a0ad69ca26c600f0f56c68dd5a1c102d4f1f/src/lib/dialog/dialog-ref.ts#L75), [stackoverflow](https://stackoverflow.com/questions/58198544/angular-dialogref-unsubscribe-do-i-need-to-unsubscribe-from-afterclosed)